src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ExecPlanGraph/Comparison/SkeletonManager.cs

```diff
@@ -22,8 +22,11 @@ public class SkeletonManager
         /// </summary>
         /// <param name="root">Node to construct skeleton of</param>
         /// <returns>SkeletonNode with children representing logical descendants of the input node</returns>
-       public SkeletonNode CreateSkeleton(Node root)
+        public SkeletonNode CreateSkeleton(Node root)
         {
+            if (root == null)
```

Comment:
If's should always have parens for the body of the if.

But also - is root being null something you actually expect? Seems like letting it throw an error in that case is preferable (although maybe throw your own exception so it's clear what the issue is)

Reply:
(same comment for all the other similar checks)

Reply:
Null isn't something I would expect. I added that because I was writing a unit for this section of the ported `SkeletonManager`'s `AreSkeletonsEquivalent` method:
![image](https://user-images.githubusercontent.com/87730006/158663511-818d31ef-e5ec-4ab9-b1a7-d2e18ab77b83.png)

The setup for the aforementioned unit test looks like this:
![image](https://user-images.githubusercontent.com/87730006/158668481-ea85dcaa-8ee8-4e87-9dbb-bea3c84652a3.png)

As seen in the unit test, it starts by setting two `Node` variables to null and then uses those `Node` variables to create `SkeletonNodes`. The null check was then added to `CreateSkeleton` to prevent a null exception from being thrown when the method accessed a property on an object that is null.

Reply:
Generally tests shouldn't be sending in invalid values. You should instead either create an actual Node object and pass it in or create a mock and use that.